packages/suite/src/support/suite/Protocol.tsx

```diff
@@ -0,0 +1,120 @@
+import { useEffect, useCallback } from 'react';
+import Router, { useRouter } from 'next/router';
+import { useActions, useSelector } from '@suite-hooks';
+import { isWeb, isDesktop } from '@suite-utils/env';
+import { isAddressValid } from '@wallet-utils/validation';
+import { getProtocolInfo } from '@suite-utils/parseUri';
+import * as sendFormActions from '@wallet-actions/sendFormActions';
+import { getDefaultValues } from '@wallet-utils/sendFormUtils';
+import { Network } from '@wallet-types';
+
+const protocolToCoin: { [key: string]: Network['symbol'] } = {
+    bitcoin: 'btc',
+};
+
+const allowedActions = [
+    'accounts/send',
+    // Add more actions
+];
+
+const Protocol = () => {
+    const { accounts, localCurrency, enabledNetworks } = useSelector(state => ({
+        accounts: state.wallet.accounts,
+        localCurrency: state.wallet.settings.localCurrency,
+        enabledNetworks: state.wallet.settings.enabledNetworks,
+    }));
+    const { getDraft, saveDraft } = useActions({
+        getDraft: sendFormActions.getDraft,
+        saveDraft: sendFormActions.saveDraft,
+    });
+
+    const createDraftForCoin = useCallback(
+        payload => {
+            const symbol = protocolToCoin[payload.coin];
+
+            // Check if the network is enabled
+            if (!enabledNetworks.includes(symbol)) return '';
+
+            // Don't continue if the address is not valid
+            if (payload.address && !isAddressValid(payload.address, symbol)) return '';
```

Comment:
Could we let the send form's validation handle invalid addresses? I think it will be less confusing for user in case he/she really clicks on bitcoin ui with invalid address.
Scenario: you click the bitcoin link, suite will open, but nothing really happen just because of invalid address and you don't know that and just blame stupid suite for not working again! 😁 

I guess the alternative is to make additional UI specially for bitcoin uri handling, we could also use some modal so we can let the user chose from which account he want to send the transaction, that would be really cool.

Reply:
The validity of the address has to be checked before creating a draft. But there's also the issue when a draft already exists. I couldn't think of a good way to solve that but I like the idea of prompting a modal to the user. This could also be used to decide on whether to dismiss the current draft in favour of the incoming one.